### ISSUE TITLE
everything

### ISSUE THREAD

**Issue by orbogink** [BranchMetrics/react-native-branch-deep-linking-attribution#620]
doesn't work. All the time. Every time. Thanks.

**Comment by Seamus1989** [comment]
+1

**Comment by AndrewMorsillo** [comment]
Yeah the state of this SDK and the documentation is... not optimal, to put it nicely. So far I've discovered the following undocumented or broken behavior:

1. getLatestReferringParams and getFirstReferringParams don't work until after the subscribe callback has fired
2. "Trouble reaching the Branch servers, please try again shortly" randomly on Android even when all other network operations in the app are working flawlessly
3. "session initialization already happened" error shows up randomly
4. "Request to branch service timed out" while all other network operations in the app work fine
5. Most of the time there are no deferred deep link parameters on Android
6. Lots of very ambiguous, under-defined, and outdated documentation


It seems like Branch doesn't actually _support_ react native given the amount of serious bugs, open issues here, and lack of activity on the repo. It's a shame because it would be really useful if it worked.

**Comment by ahmednawar** [comment]
I manage the SDK team at Branch and wanted to chime in regarding the state of the RN SDK.

We invested a lot of effort in building this SDK because we wanted to build a world class SDK for the React Native community. Sadly, we didn't sustain the same level of investment over the past year and the quality dropped. Our team is pretty small and was spread very thin between iOS 14 changes and other internal priorities. The SDK was a casualty of some tough choices we had to make.

We are committed to supporting React Native and we want to be proud of our SDKs. I am working with the team and the Branch leadership to invest more resources into our SDKs and we are hoping to triple the size of the team before the end of the year. 🤞 for some good news soon.

**Comment by ahmednawar** [comment]
We've opened a new position to maintain the React Native SDK and other JS Plugins. If anyone is interested, please apply here or forward to anyone who might be https://branch.io/careers/apply/?gh_jid=3360544
